sabnzbd/deobfuscate_filenames.py

```diff
@@ -225,90 +232,140 @@ def deobfuscate(nzo, filelist: List[str], usefulname: str):
             logging.info("Deobfuscate renaming %s to %s", otherfile, new_name)
-            # Rename and make sure the new filename is unique
-            renamer(otherfile, new_name)
+            filtered_filelist.remove(otherfile)
+            filtered_filelist.append(renamer(otherfile, new_name))
             nr_files_renamed += 1
 
     if nr_files_renamed:
         nzo.set_unpack_info("Deobfuscate", T("Deobfuscate renamed %d file(s)") % nr_files_renamed)
+
+    return filtered_filelist
+
+
+def without_extension(fullpathfilename: str) -> str:
+    """Returns full file path, without extension
+    So '/some/dir/somefile.bin' results in '/some/dir/somefile'"""
+    return os.path.splitext(fullpathfilename)[0]
+
+
+def deobfuscate_subtitles(filelist: List[str]):
+    """
+    input must be a list of existing filenames
+
+    Find .srt subtitle files, and rename to match the biggest file (if there is a clearly biggest file)
+
+    Some_Big_File_2024.avi      # biggest file
+    Some_Big_File_2024.srt      # no renaming wanted
+    Some_Big_File_2024.ger.srt  # no renaming wanted
+    14_English.srt              # to be renamed
+    dut.srt                     # to be renamed
+    Something.else.txt          # no renaming wanted, because no .srt
+
+    will result in
+
+    Some_Big_File_2024.avi
+    Some_Big_File_2024.srt
+    Some_Big_File_2024.ger.srt
+    Some_Big_File_2024.14.English.srt   # renamed by prepending base name (and replacing _)
+    Some_Big_File_2024.dut.srt          # renamed by prepending base name
+    Something.else.txt
+
+    """
+
+    # find .srt files
+    if not (srt_files := [f for f in filelist if f.endswith(".srt")]):
+        logging.debug("No .srt files found, so nothing to do")
+        return None
+
+    # check there is a clearly biggest file
+    if not (biggest_file := get_biggest_file(filelist)):
+        logging.debug("No clearly biggest file found, so no subtitle renaming feasible")
+        return None
+
+    biggest_file_without_ext = without_extension(biggest_file)  # get full path base name of biggest file
+    logging.debug(f"Using as base filename: {biggest_file_without_ext}")
+
+    # handle srt files one by one
+    for srt_file in srt_files:
+        if without_extension(srt_file).startswith(biggest_file_without_ext):
+            # already the same start as the biggest file, so skip
+            continue
+        # not the same start, so rename the srt file
+        filename_only = os.path.basename(srt_file)  # like "14_English.srt", so without path
+        filename_only = filename_only.replace("_", ".")  # replace underscore with dot; better for VLC
+        # now put that name after the base name of the biggestfile:
+        new_full_name = f"{biggest_file_without_ext}.{filename_only}"  # put (renamed) srt behind that
+        unique_filename = get_unique_filename(new_full_name)  # make sure it's really unique
+        renamer(srt_file, unique_filename)  # ... and rename actual file on disk
```

Comment:
Should we maybe also include a History update?
```
nzo.set_unpack_info("Deobfuscate", T("Deobfuscate renamed %d subtitle file(s)") % nr_files_renamed)
```

Reply:
Good idea. Done